### ISSUE TITLE
Add logical rep shard transfermode that falls back to blocking

### ISSUE THREAD

**Issue by JelteF** [citusdata/citus#6755]
For UNLOGGED tables `force_logical` does not work because these tables cannot be added to replication slots. Right now the only way around this is to either remove the UNLOGGED table from the colocation group or to use `block_writes`. It would be much nicer if tables that supported it could still use logical replication, but only UNLOGGED tables would have be transferred in `block_writes` mode.

Related: #6374 and https://github.com/citusdata/citus-enterprise/issues/394

**Comment by marcocitus** [comment]
Rather than phrasing this as a transfer mode, probably we should just do always do something reasonable for unlogged tables?

e.g. not copying any data could even be reasonable

**Comment by JelteF** [comment]
Yeah, that's why I made this a separate issue from #6374. This one is very related, but orthogonal to the UNLOGGED table situation. The idea is (which I realise I did not write down clearly in the issue description) is that this option would also be used for other tables that don't have a replica identity. This could be useful e.g. for replicating certain small reference tables that don't have a primary key.
